src/DynamoDb.DistributedLock/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -16,25 +17,30 @@ public static class ServiceCollectionExtensions
     /// <param name="services">The service collection to register with.</param>
     /// <param name="configuration">The configuration source.</param>
     /// <param name="sectionName">The name of the configuration section to bind to <see cref="DynamoDbLockOptions"/>.</param>
+    /// <param name="awsOptionsSectionName">The name of the configuration section to be read with GetAWSOptions</param>
     /// <returns>The updated <see cref="IServiceCollection"/>.</returns>
     public static IServiceCollection AddDynamoDbDistributedLock(this IServiceCollection services,
         IConfiguration configuration,
-        string sectionName = DynamoDbLockOptions.DynamoDbLockSettings)
+        string sectionName = DynamoDbLockOptions.DynamoDbLockSettings, string? awsOptionsSectionName = null)
```

Comment:
Adding these optional parameters breaks binary compatibility, but because this has an existing optional param of the same type, I couldn't think of a way to do the trick of manually making another set of methods, because they are ambiguous and one gets hidden.

I think this is probably OK, since in my experience binary breaks are not as big of a deal in modern dotnet + nuget, and if you did hit this by copying some old dll, you would see it immediately in startup.